modal/image.py

```diff
@@ -508,12 +513,22 @@ def _from_args(
         if build_function and len(base_images) != 1:
             raise InvalidError("Cannot run a build function with multiple base images!")
 
+        validated_volumes = validate_volumes(volumes)
+        validated_volumes_: list[tuple[str, _Volume]] = []
+
+        for k, v in validated_volumes:
+            if isinstance(v, _CloudBucketMount):
+                raise InvalidError("Image builds only support mounting modal.Volume")
```

Comment:
Dang, I feel like we just plugged this gap in `.with_options` and now there's a new one :(

Reply:
Super minor but regarding the error message language, `.run_function` _does_ support CloudBucketMount. (Also, potentially more important: I'm not sure if we would end up here with a `.run_function` layer but any chance this would break that?)